bundles/binding/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/protocol/SatelMessage.java

```diff
@@ -0,0 +1,179 @@
+		// we need at least command and checksum
+		if (buffer.length < 3) {
+			logger.error("Invalid message length: {}", buffer.length);
+			return null;
+		}
+
+		// check crc
+		int receivedCrc = 0xffff & ((buffer[buffer.length - 2] << 8) | (buffer[buffer.length - 1] & 0xff));
+		int expectedCrc = calculateChecksum(buffer, buffer.length - 2);
+		if (receivedCrc != expectedCrc) {
+			logger.error("Invalid message checksum: received = {}, expected = {}", receivedCrc, expectedCrc);
+			return null;
+		}
+
+		SatelMessage message = new SatelMessage(buffer[0], new byte[buffer.length - 3]);
+		if (message.payload.length > 0)
+			System.arraycopy(buffer, 1, message.payload, 0, buffer.length - 3);
+		return message;
+	}
+
+	/**
+	 * Returns command byte.
+	 * 
+	 * @return the command
+	 */
+	public byte getCommand() {
+		return this.command;
+	}
+
+	/**
+	 * Returns the payload bytes.
+	 * 
+	 * @return payload as byte array
+	 */
+	public byte[] getPayload() {
+		return this.payload;
+	}
+
+	/**
+	 * Returns the message serialized as array of bytes with checksum calculated
+	 * at last two bytes.
+	 * 
+	 * @return the message as array of bytes
+	 */
+	public byte[] getBytes() {
+		byte buffer[] = new byte[this.payload.length + 3];
+		buffer[0] = this.command;
+		if (this.payload.length > 0)
+			System.arraycopy(this.payload, 0, buffer, 1, this.payload.length);
+		int checksum = calculateChecksum(buffer, buffer.length - 2);
+		buffer[buffer.length - 2] = (byte) ((checksum >> 8) & 0xff);
+		buffer[buffer.length - 1] = (byte) (checksum & 0xff);
+		return buffer;
+	}
+
+	private String getPayloadAsHex() {
+		StringBuilder result = new StringBuilder();
+		for (int i = 0; i < this.payload.length; ++i) {
+			if (i > 0)
+				result.append(" ");
+			result.append(String.format("%02X", this.payload[i]));
+		}
+		return result.toString();
+	}
+
+	/**
+	 * Calculates a checksum for the specified buffer.
+	 * 
+	 * @param buffer
+	 *            the buffer to calculate.
+	 * @return the checksum value.
+	 */
+	private static int calculateChecksum(byte[] buffer, int length) {
+		int checkSum = 0x147a;
+		for (int i = 0; i < length; i++) {
+			checkSum = ((checkSum << 1) | ((checkSum >> 15) & 1));
+			checkSum ^= 0xffff;
+			checkSum += ((checkSum >> 8) & 0xff) + (buffer[i] & 0xff);
+		}
+		checkSum &= 0xffff;
+		logger.trace("Calculated checksum = {}", String.format("%04X", checkSum));
+		return checkSum;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public String toString() {
+		return String.format("Message: command = %02X, payload = %s", this.command, getPayloadAsHex());
+	};
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public boolean equals(Object obj) {
+		if (obj == null)
+			return false;
+
+		if (!obj.getClass().equals(this.getClass()))
+			return false;
+
+		SatelMessage other = (SatelMessage) obj;
+
+		if (other.command != this.command)
+			return false;
+
+		if (!Arrays.equals(other.payload, this.payload))
+			return false;
+
+		return true;
+	}

```

Comment:
generally overriding equals() also requires overriding hashCode() as well (see http://docs.oracle.com/javase/7/docs/api/java/lang/Object.html#equals(java.lang.Object)). Are you sure overriding hashCode() is not necessary in this case?